week-1-ts-ecosystem/w01e03-object-oriented-structure/src/product/models/product-properties.model.ts

```diff
@@ -0,0 +1,7 @@
+export type ProductId = string;
+
+export interface ProductProperties {
+    name: string;
+    amount: number;
+    price?: number;
+}
```

Comment:
W mojej opinii ten `interface` jest nadmiarowy.

W sensie @zajewskp , rozumiem Twoją koncepcję, bo używasz podobnych wartości początkowych w konstruktorach, ale ja bym je podał implicit - w tych konstruktorach.

I tak masz ślicznie tutaj przygotowaną hierarchię klas - ładnie u Ciebie widać dokładnie jakimi modelami danych operujesz.
Więc to już jest za bardzo `DRY` moim zdaniem :) 

Tutaj można by właśnie w konstruktorach pokazać czym różnią się od siebie auction-product i buy-now od `free-` i to jest ok.